components/org.wso2.carbon.identity.mgt.store.connector.jdbc/src/main/java/org/wso2/carbon/identity/mgt/store/connector/jdbc/queries/MySQLFamilySQLQueryFactory.java

```diff
@@ -279,4 +282,28 @@ public MySQLFamilySQLQueryFactory() {
         sqlQueries.put(ConnectorConstants.QueryTypes.SQL_QUERY_UPDATE_PASSWORD_INFO, UPDATE_PASSWORD_INFO);
         sqlQueries.put(ConnectorConstants.QueryTypes.SQL_QUERY_DELETE_CREDENTIAL, DELETE_CREDENTIAL);
     }
+
+    public String getQuerryForUserIdFromMultipleAttributes(List<Attribute> attributes, int offset, int length) {
+        StringBuilder getUniqueUserQuerry = new StringBuilder();
+         getUniqueUserQuerry.append("SELECT UM_USER.USER_UNIQUE_ID FROM UM_USER WHERE UM_USER.ID IN");
+        int count = 1;
+        for (Attribute attribute : attributes) {
+            getUniqueUserQuerry
+                    .append(" (SELECT UM_USER_ATTRIBUTES.USER_ID FROM UM_USER_ATTRIBUTES" +
+                            " WHERE ATTR_ID = (SELECT ID FROM UM_ATTRIBUTES WHERE ATTR_NAME = '")
+                    .append(attribute.getAttributeName())
+                    .append("' ) AND ATTR_VALUE = '")
+                    .append(attribute.getAttributeValue())
```

Comment:
Is there a validation happening for these attribute values and names list. With the way the query is built, it seems SQL injection attacks are possible, if no previous validation has happened. Please confirm.

Reply:
Need to verify and change